test/README.md

```diff
@@ -1,95 +1,296 @@
+"metricsValidatorImage": "localhost:5000/microsoft/metrics-validator:latest-linux-amd64",
+"caCertScriptPath": "/home/azureuser/iotedge/test/certs",
+"dpsIdScope": "0ne003A5268",
+"rootCaCertificatePath": "/home/azureuser/iotedge/test/certs/certs/azure-iot-test-only.root.ca.cert.pem",
+"rootCaPrivateKeyPath": "/home/azureuser/iotedge/test/certs/private/azure-iot-test-only.root.ca.key.pem",
+"verbose": "true",
+"logFile": "/home/azureuser/iotedge/logs/logfile",
+"registries":
+        [
+                {
+                        "address": "localhost:5000",
+                        "username": "username"
+                }
+        ]
+}
+~~~
+
+### Test secrets
+
+The tests also expect to find several _secret_ parameters. While these can technically be added to `context.json`, it is recommended that you create environment variables and make them available to the test framework in a way that avoids committing them to your shell's command history or saving them in clear text on your filesystem. When set as environment variables, all secret parameters must be prefixed with `E2E_`. Parameter names are case-**in**sensitive; they're only shown in uppercase here to follow the common convention for environment variables, and to stand out as secrets.
+
+| Name | Required | Description |
+|------|----------|-------------|
+| `E2E_DPS_GROUP_KEY` | * | The symmetric key of the DPS [enrollment group](https://docs.microsoft.com/en-us/azure/iot-dps/concepts-service#enrollment-group) to use. Used when running any DPS tests, ignored otherwise. |
+| `E2E_IOT_HUB_CONNECTION_STRING` | ✔ | Hub-scoped IoT Hub connection string that can be used to get/add/remove devices, deploy edge configurations, and get/update module twins. |
+| `E2E_EVENT_HUB_ENDPOINT` | ✔ | Connection string used to connect to the Event Hub-compatible endpoint of your IoT Hub, to listen for D2C events sent by devices or modules. |
+| `E2E_REGISTRIES__{n}__PASSWORD` || Password associated with a container registry entry in the `registries` array of `context.json`. `{n}` is the number corresponding to the (zero-based) array entry. For example, if you specified a single container registry in the `registries` array, the corresponding parameter would be `[E2E_]REGISTRIES__0__PASSWORD`. |
+| `E2E_ROOT_CA_PASSWORD` || The password associated with the root certificate specified in `rootCaCertificatePath`. |
+| `E2E_BLOB_STORE_SAS` || The sas token used to upload module logs and support bundle in the tests. |
+
+_Note: the definitive source for information about test parameters is `test/Microsoft.Azure.Devices.Edge.Test/helpers/Context.cs`._
+
+#### Sample environnment settings
+~~~ sh
+export E2E_IOT_HUB_CONNECTION_STRING="HostName=..."
+export E2E_EVENT_HUB_ENDPOINT="Endpoint..."
+export E2E_DPS_GROUP_KEY="IGcMo..."
+export DPS_GROUP_KEY="IGcMoH..."
+export E2E_REGISTRIES__0__PASSWORD="password"
+export E2E_BLOB_STORE_SAS="https://...."
+~~~

```

Comment:
I wonder if we can script a really simple script that a user can execute so that the script will ask and source the password for them. That way there is no need to store a password in a plain text.

Reply:
I have taken the details on how to set the environment out of the README and left it to the developer's discretion.